src/app/execution/txFormat.js

```diff
@@ -88,7 +92,11 @@ module.exports = {
         dataHex = bytecodeToDeploy + dataHex
       }
     } else {
-      dataHex = Buffer.concat([helper.encodeFunctionId(funAbi), data]).toString('hex')
+      if (params.startsWith("0x")) {
+        dataHex = params;
+      } else {
+        dataHex = Buffer.concat([helper.encodeFunctionId(funAbi), data]).toString('hex')
```

Comment:
@yann300 -- I'm not sure of this check here though. I've added it just to avoid the access of `data` variable because it'd be empty string if param is a hex encoded value as per the logic above

Reply:
Well then in your case the input given by the user has to contain the function id.
I think for a first commit, we should stick to the fact that `params` will only contain the encoded input params and not the function id . We could be more fine-grain later on if necessary.
So from `dataHex` , I am sure we can generate `data`.

Reply:
Sure. Let me try that and push the change. Thanks.

Reply:
@eswarasai  is everything ok with that PR, do yo need something?

Reply:
@yann300 -- Sorry for the delay in getting back to you. Yeah, so about getting the `data` from the `dataHex` is essentially not required as we won't have the `functionId`. I guess this PR should handle the requirement. Let me know a way to test it out and confirm that everything works as expected. Thanks.

Reply:
you do have the `functionId` at this point (it is part of `funabi`)
The actual state of the PR assumes that the hex value you input contains the function id.
you should assumes that the hex value you input only contain the encoded data (there's room for improvement but that's another PR)
to test it, you can have:

```
contract test {
    function t (uint _p) constant returns (uint) {
        return _p;
    }
}
```
call `t ()` with the following input
`"0x000000000000000000000000000000000000000000000000000000000000000a"`
the return value should be `10`

Reply:
@yann300 -- I've updated the code now. Below are the modifications :
1) Added specific check to see if the `params` value specifically `startsWith` `"0x`. Let me know is this needs to be changed to `contains` - `0x`
2) Converting the input param to `dataHex`
3) Getting `data` from the `dataHex`

I've used the above test scenario and output always seemed to be `uint256: 10` with and without `0x` input. Let me know if there's any other modification required. Thanks.